### PR TITLE
dont apply custom filters per plugin

### PR DESCRIPTION
**Pull request by stevenctl**
# Description

This code should not have been called in a loop. It ended up there because it used to be its own plugin.
The test has been updated to check the exact count. 

# Checklist:

- [X] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/9915

### CODE REVIEW

File: changelog/v1.18.0-beta16/custom-filters-loop.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NON_USER_FACING
```
Comment (EItanya) [inline changelog/v1.18.0-beta16/custom-filters-loop.yaml:2]: Nit: This is technically a bug fix, can you switch it and use the original issue as the `issueLink`?

Comment (stevenctl) [reply changelog/v1.18.0-beta16/custom-filters-loop.yaml]: Done. Does this mean a user would see this changelog? They probably won't even get a chance to see the problem crop up.

**Review by EItanya — APPROVED** [review]

**Review by npolshakova — APPROVED** [review]

**Comment by npolshakova** [comment]
Looks good, just need to move the changelog

**Comment by stevenctl** [comment]
It's a custom tag in my repo breaking this. Deleted it on my remote hopefully it passes now.

**Review by EItanya — APPROVED** [review]

**Pull request merged**